cores/esp8266/core_esp8266_waveform_pwm.cpp

```diff
@@ -157,9 +156,10 @@ static IRAM_ATTR void disableIdleTimer() {
 // Wait for mailbox to be emptied (either busy or delay() as needed)
 static IRAM_ATTR void _notifyPWM(PWMState *p, bool idle) {
   p->pwmUpdate = nullptr;
+  MEMBARRIER();
   pwmState.pwmUpdate = p;
   MEMBARRIER();
-  forceTimerInterrupt();
+  // The ISR pulls updates on the next PWM interval
```

Comment:
That could be 10 **milli**seconds away.  forceTimerInterrupt would make it happen in 10 **micro**seconds.  I think this needs to be undone.

Reply:
I think: the ISR does NOT act on the request immediately when the ISR is triggered, but only as scheduled previously, anyway, because of `!pwmState.cnt`:
```
  } else if (!pwmState.cnt && pwmState.pwmUpdate) {
    // Start up the PWM generator by copying from the mailbox
    pwmState.cnt = 1;
    pwmState.idx = 1; // Ensure copy this cycle, cause it to start at t=0
    pwmState.nextServiceCycle = GetCycleCountIRQ(); // Do it this loop!
    // No need for mem barrier here.  Global must be written by IRQ exit
  }
```

Reply:
So, it's a tradeoff, when it's the first PWM starting, you're right, if anything else is running already, it's wasting resources and may very badly affect PWM/waveform timings.